src/crimson/net/SocketMessenger.h

```diff
@@ -152,6 +152,18 @@ class SocketMessenger final : public Messenger {
   Interceptor *interceptor = nullptr;
 #endif
 
+  seastar::future<> mark_down(const entity_addr_t& a) final {
+    auto conn = lookup_conn(a);
+    if (conn) {
+      return seastar::smp::submit_to(
+	conn->get_shard_id(),
+	[conn] {
+	conn->mark_down();
+	return seastar::now();
+      });
+    }
```

Comment:
`conn` is of type `SocketConnectionRef`, which isn't a foreign ptr so shouldn't be captured/copied by a different shard.

How about:
```c++
return seastar::smp::submit_to(
  conn->get_shard_id(),
  [conn=conn.get()] {
  conn->mark_down();
  return seastar::now();
}).then([conn] {});
```
?

Reply:
Can we move it and reset `conn` afterwards?

Reply:
Please make as needs-crimson-qa once addressed